src/bentoml/_internal/service/openapi/utils.py

```diff
@@ -1,15 +1,47 @@
 from __future__ import annotations
 
+import sys
 import typing as t
 from typing import TYPE_CHECKING
 from functools import lru_cache
 
+if sys.version_info >= (3, 10):
+
+    from types import UnionType
+
+    def is_union(tp):  # check Union[A,B] or A | B
+        return tp == t.Union or get_origin(tp) == UnionType
+
+else:
+
+    def is_union(tp):  # check Union[A,B]
+        return tp == t.Union
+
+
+if sys.version_info >= (3, 11):
+    from typing import get_args
+    from typing import Required
+    from typing import TypedDict
+    from typing import get_origin
+    from typing import NotRequired
+    from typing import is_typeddict
+    from typing import get_type_hints
+else:
+    from typing_extensions import get_args
+    from typing_extensions import Required
+    from typing_extensions import TypedDict
+    from typing_extensions import get_origin
+    from typing_extensions import NotRequired
+    from typing_extensions import get_type_hints
+    from typing_extensions import is_typeddict
+
```

Comment:
Let's move this to `bentoml/_internal/types.py`. Also there is already some backport function that we include there that doesn't rely on typing_extensions.

@sauyon if we decided to introduce typing_extensions, then maybe we can also cleanup `bentoml/_internal/types.py`?

Reply:
I think it's enough to move them under `if TYPE_CHECKING` block, we can assume the version is the latest when we work with types.

Reply:
@frostming I'm thinking about `bentoml._internal.types` as a compat typing layer to `python` typing